### DISCUSSION TITLE
Problem loading german onnx model

### DISCUSSION THREAD

**Discussion by TheRealSvc** [snakers4/silero-models#46] [Q&A] [answered]
Please Support: I have tried both the english pytorch model and the onnx model. Both work fine on different wav files i've tried.  After that i tried the same for the german models. Again the pytorch model is fine, but the onnx model seems totally off. So my idea was to export to create a new onnx model from the pytorch-model using "torch.onnx.export(...)", but unfortunately this fails with error: 

> _temporary: the only valid use of a module is looking up an attribute but found  = prim::SetAttr[name="num_batches_tracked"](%4257, %4387)_

I was looking for a fix but could'nt find any clue on how to make this work. Any ideas here ?

**Comment by snakers4** [comment]
> Again the pytorch model is fine, but the onnx model seems totally off.

In theory they should have roughly similar outputs.
Can you please show an example how to replicate this error?

**Comment by snakers4** [comment]
![image](https://user-images.githubusercontent.com/12515440/106572453-7bdc8680-6549-11eb-9494-126e5931a11a.png)

**Comment by snakers4** [comment, marked as answer]
My theory is that you forgot to change `en` to `de` in this line

`torch.hub.download_url_to_file(models.stt_models.de.latest.onnx, 'model.onnx', progress=True)`

**Comment by TheRealSvc** [comment]
Yes, your theory was correct. I stupidly assumed that by setting the language at the top it should be set everywhere it occurs.  Thank you.

**Comment by snakers4** [comment]
Well, this could have been prevented
I assume that you can do something like

```
torch.hub.download_url_to_file(models.stt_models.get(language, 'en').latest.onnx, 'model.onnx', progress=True)
```
In any case, a PR with minor fixes like this would be appreciated
